Assets/Scripts/Interfaces/IDamageable.cs

```diff
@@ -0,0 +1,11 @@
+using UnityEngine;
+
+public interface IDamageable
+{
+    public float Health { set; get; }
+    public bool Targetable { set; get; }
+    public bool Invincible { set; get; }
+    public void OnHit(float damage, Vector2 knockback);
```

Comment:
It's not clear what `knockback` represents. Is it the source of the damage, the direction, or what? I initially thought it was the knockback strength (which could be nice to add separately), but it doesn't fit the type.

Reply:
it's a vector2 of direction + strength in which you're knocked back.